### DISCUSSION TITLE
validation for email regex including empty strings

### DISCUSSION THREAD

**Discussion by yeoldegrove** [Budibase/budibase#3117] [Help] [answered]
I am trying to validate an email address in a Form Field. The email validation regex part and the single space (`^ $`) parts work fine....
however the empty string part does just not work and keeps complaining after the empty input --> `^$`

"Custom validation rules"
"Must match regex"
"Value"
`^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+.[a-zA-Z]{2,6}$|^$|^ $`

Any ideas on how to achieve my goal?

**Comment by mjashanks** [comment, marked as answer]
Hi Eike. I found this one to work: 
```
[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?
```

**Reply by NicolasFabre81** [reply]
I'm afraid it is not perfect : for example, zz@zz.com' does validate

**Reply by shogunpurple** [reply]
Are you sure? It doesn't seem to:
![Screenshot 2023-01-03 at 10 43 51](https://user-images.githubusercontent.com/11256663/210342090-8c0743ff-5999-4d96-bae4-6365d168f3b9.png)

Also tested inside BB and working as expected.

**Reply by NicolasFabre81** [reply]
Yes, tested again in BB V2.1.42 : I manage to save the row with the wrong email zz@zz.com' whereas the control is enabled : 
<img width="499" alt="image" src="https://user-images.githubusercontent.com/28592069/210588529-529ab13c-a048-4067-9a50-e40d0decd009.png">

**Reply by NicolasFabre81** [reply]
Sorry guys, but in V2.4 the problem remains : quotes ' are not filtered by this regex

**Comment by yeoldegrove** [comment]
Your email regex is much more sophisticated but only covers the email part.
"empty input" is really my problem here... which should also be valid.

**Reply by yeoldegrove** [reply]
Appending `|^$|^ $` to your regex does also not work for me.

**Reply by mjashanks** [reply]
Ah-ha. I get you...

If you use "Configure Condition" (bottom right), then you can apply the validation only when the field is set....

![image](https://user-images.githubusercontent.com/3524181/138437783-60449f4c-dca6-4d5a-bb59-56f92fc2ad7e.png)

**Comment by zeigerpuppy** [comment]
How about this based on RFC 5322

```regex
^((?:[A-Za-z0-9!#$%&'*+\-\/=?^_`{|}~]|(?<=^|\.)"|"(?=$|\.|@)|(?<=".*)[ .](?=.*")|(?<!\.)\.){1,64})(@)((?:[A-Za-z0-9.\-])*(?:[A-Za-z0-9])\.(?:[A-Za-z0-9]){2,})$
```

https://regex101.com/r/3uvtNl/1
